File: registration/src/schema/student.js
import JOI from 'joi-browser'

export const StudentSchema = {
    firstName:JOI.string()
        .required()
        .label('First Name'),
    lastName:JOI.string()
        .required()
        .label('Last Name'),
    phoneNo:JOI.string()
        .required()
        .regex(/^[0-9]{10}$/)
        .error(() => {
            return {
                message: 'Phone no is required and should be 10 digits',
            };
        })
        .label('Phone'),
    email:JOI.string()
        .required()
        .email()
        .label('Email'),
    nic:JOI.string()
        .required()
        .regex(/^[0-9]{12}$/)
        .error(() => {
            return {
                message: 'NIC is required and should be 12 digits',
            };
        })
        .label('NIC'),
    address:JOI.string()
        .required()
        .label('Address'),
    dateOfBirth:JOI.date()
        .required()
        .label('Date of Birth'),
    avatar:JOI.allow()
    
   
}

File: StudentManagement - backend/StudentManagementAPI/Models/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StudentManagementAPI.Models
{
    public class Student
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNo { get; set; }
        public string Email { get; set; }
        public string NIC { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string Address { get; set; }
        public string Avatar { get; set; }

     


    }
}
